### PR TITLE
Week 2 n dhruva

### PR DESCRIPTION
**Pull request by ndhruv03**
Screenshot from 2022-12-31 10-31-36
Screenshot from 2022-12-31 10-33-23
Screenshot from 2022-12-31 10-36-36

### CODE REVIEW

**Review by hrushikeshj — APPROVED** [review]
Good job.
Looks like you have branched from week1 branch, next time make sure to branch from `main`

File: week_2/prime_numbers.rb [hunk 1]
```diff
@@ -14,5 +14,29 @@
...
-  raise NotImplementedError # TODO
+  if n < 0
+    raise ArgumentError.new("Negative Input is not allowed!")
+  end
+  
+  ans = Array.new()
+  index = 2
+  index2 = 2
+  flag = true
+  while index <= n
```
Comment (hrushikeshj) [inline week_2/prime_numbers.rb:25]: `each` or `for` loop can also be used.

```ruby
(2..n).each do |num|
  # ...
end
```

Comment (hrushikeshj) [inline week_2/prime_numbers.rb:25]: `each` or `for` loop can also be used. [Ruby ranges](https://www.rubyguides.com/2016/06/ruby-ranges-how-do-they-work/)

```ruby
(2..n).each do |num|
  # ...
end
```

Comment (ndhruv03) [reply week_2/prime_numbers.rb:25]: Thanks. will look into them too

**Comment by ndhruv03** [comment]
Yes i just realised after you told that i bhad branched from week1, will ensure to branch from main
 next time around

**Pull request closed without merging**